artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ChannelImpl.java

```diff
@@ -511,22 +511,24 @@ public void close() {
    public void transferConnection(final CoreRemotingConnection newConnection) {
       // Needs to synchronize on the connection to make sure no packets from
       // the old connection get processed after transfer has occurred
-      synchronized (connection.getTransferLock()) {
-         connection.removeChannel(id);
+      if (connection != null) {
```

Comment:
same thing here.. how connection would be null?

Reply:
Well, it is null-checked few lines bellow so it confued me.